### DISCUSSION TITLE
Temabygger

### DISCUSSION THREAD

**Discussion by mrosvik** [digdir/designsystemet#1872] [Theming]
Verktøy for å forstå hvordan fargesystemet fungererer, samt generere egne fargeskalaer.

⚠️ **Verktøyet er under arbeid** ⚠️ 
- På forsiden kan du velge en `accent`-farge og få en forhåndsvisning av komponentene. 
- Under "Liste over fargeskalaer" kan du se flere farger med mer informasjon om kontrast.

![image](https://github.com/digdir/designsystemet/assets/1464915/f253f5c0-c596-4723-b9de-8a59b9690afa)

Foreløpig demo: https://thuneer.github.io/designsystemet-colors/

Meld inn behov til verktøyet i denne tråden 🧵

**Comment by Camulos** [comment]
Behov, kombiner med: https://github.com/digdir/designsystemet/discussions/1873

Andre mulige behov:

- Mulighet for å mellomlagre farger som er valgt mens en jobber
- laste ned i json/css/ts eller liknende med fargesystem, i standardisert oppsett
- laste opp json/css/ts eller liknende med eget fargesystem
- Legge inn på forhånd fargesystemer fra offentlige virksomheter som har tatt DS ibruk og som har definert eget fargesystem (legge inn en mulighet for å velge disse utover et "nøytralt" tema/fargesystem
- vise både mørk og lys modus samtidig eller gi en enkel switch på siden
- kopier fargekode-knapp
- mulighet for å legge inn farger med flere metoder, ikke bare picker og HEX
- Guide: hvordan bruke fil fra dette verktøyet for å overstyre globalt hos seg selv det nøytrale fargesystemet
- Guide: meningsbærende farger foreslås fra DS at ikke bør endres, men om de likevel ønsker å gjøre det så... hvordan?
- Guide: hvordan en kan bruke saturation uten å "ødelegge" kontraststegene i fargesystemet
- Valg: mulighet for å kun tillate farger fra fargesystemets kontraststeg, altså ingen små justeringer på L

Eksempel standardisert fil med fargesystem (ikke funksjonelle farger, kun "grunnsystemet"): https://github.com/brreg/br-contrast/blob/main/data/colors.ts

**Reply by Camulos** [reply]
Mulighet for å manuelt sette hver enkelt globale og deretter funksjonelle farge, med veiledning
